.github/workflows/pr.yml

```
@@ -0,0 +1,58 @@
+name: Test Pull Request
+
+on: [pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8']
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache .cache/pip
+      uses: actions/cache@v3
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: pip_cache_py_${{ matrix.python-version }}
+    - name: Install package
+      run: pip install -r requirements.txt black==23.3 flake8==6.0
+    - name: Flake8
+      run: flake8 --ignore=C901,W503
+    - name: Black
+      uses: psf/black@23.3.0
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8']
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache .cache/pip
+      uses: actions/cache@v3
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: pip_cache_py_${{ matrix.python-version }}
+    - name: install package
+      run: python -m pip install .
+    - name: install packages for testing
+      run: pip install nose xmldiff
+      # TODO: replace with tox tests from #38
+      #- name: run nosetests
+      #  run: nosetests --with-coverage --cover-package=galaxyxml
+
+    - name: Run smoke test on example
+      run: |
+        python examples/example.py > tmp.xml
+        xmldiff tmp.xml examples/tool.xml
```

While testing the other PR I had the impression that xmldiff just prints the diff but does not fail on a difference. So maybe 

```suggestion
        if xmldiff tmp.xml examples/tool.xml | grep '^'; then
            exit 1
        fi
```

(untested)

---

Fixed with `--check` flag, how convenient